workers/unity/Assets/Gdk/Core/Systems/SpatialOSSendSystem.cs

```diff
@@ -44,15 +37,14 @@ private void GenerateComponentGroups()
             }
         }
 
-        public void RegisterCustomReplicationSystem(ComponentType type)
+        public bool TryRegisterCustomReplicationSystem(ComponentType type)
         {
             if (!registeredReplicators.Contains(type.TypeIndex))
             {
-                Debug.LogWarningFormat(Warnings.CustomReplicationSystemAlreadyExists, type);
-                return;
+                return false;
             }
 
-            registeredReplicators.Remove(type.TypeIndex);
+            return registeredReplicators.Remove(type.TypeIndex);
```

Comment:
Why is the `Register` method *removing* a replication system?

Reply:
From my understanding of the code if you want to use a custom replicator you need to remove the default one.

Reply:
When you register a custom replicator, it removes the default replicator from the list that it iterates over every tick.

Reply:
This probably warrants a comment, because it looks very wrong.